src/deploy-cromwell-on-azure/KubernetesManager.cs

```diff
@@ -195,13 +195,24 @@ public async Task UpdateHelmValuesAsync(IStorageAccount storageAccount, string k
         public async Task UpgradeValuesYamlAsync(IStorageAccount storageAccount, Dictionary<string, string> settings, List<MountableContainer> containersToMount)
         {
             var values = KubernetesYaml.Deserialize<HelmValues>(await Deployer.DownloadTextFromStorageAccountAsync(storageAccount, Deployer.ConfigurationContainerName, "aksValues.yaml", cts));
+            var previousVersion = new Version(values.Config.ContainsKey("cromwellOnAzureVersion") ? (string)values.Config["cromwellOnAzureVersion"] : "4.0");
             UpdateValuesFromSettings(values, settings);
             MergeContainers(containersToMount, values);
+            UpdateValues(values, previousVersion);
             var valuesString = KubernetesYaml.Serialize(values);
             await File.WriteAllTextAsync(TempHelmValuesYamlPath, valuesString);
             await Deployer.UploadTextToStorageAccountAsync(storageAccount, Deployer.ConfigurationContainerName, "aksValues.yaml", valuesString, cts.Token);
         }
 
+        private void UpdateValues(HelmValues values, Version previousVersion)
```

Comment:
Seems like this method should be named something like "UpdateContainerValues", and then in the Deployer.cs, there is a version check in the Update area, which would result in calling this.  That way it's very clear what happens during an update, not buried in KubernetesManager.cs

Reply:
My intention was that all version related updates to the helm values would be added to this method not just this example for containers. I renamed the method to ProcessHelmValuesUpdates though, and passed the installedVersion from the deployer so hopefully its a little clearer.